activerecord/lib/active_record/associations/preloader/association.rb

```diff
@@ -114,8 +114,16 @@ def reflection_scope
             @reflection_scope ||= reflection.scope_for(klass)
           end
 
+          def klass_scope
+            if (current_scope = klass.current_scope) && current_scope.empty_scope?
```

Comment:
`current_scope` is not a simple accessor. So I avoided extra `current_scope` access.
https://github.com/rails/rails/blob/v5.1.2/activerecord/lib/active_record/scoping.rb#L14

Do you prefer `klass.current_scope && klass.current_scope.empty_scope?`?